server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -835,11 +835,12 @@ private void updatePayload(final Handle handle, final DataSegment segment) throw
 
   public List<DataSegment> getUnusedSegmentsForInterval(final String dataSource, final Interval interval)
```

Comment:
I think we should use the MetadataSegmentManager in the indexing service as well, so it covers all the segment interaction with the metadata store.